src/Text/Pandoc/Writers/Markua.hs

```diff
@@ -0,0 +1,501 @@
+    return $ contents <> cr
+
+blockToMarkua opts (Para inlines) = do
+    contents <- inlineListToMarkua opts inlines
+    -- set explicit a blankline before and after para to be conform with Markua
+    return $ blankline <> contents <> blankline
+
+blockToMarkua opts (LineBlock inlines) = do -- TODO nowrap richtig?
+    contents <- mconcat <$> mapM (inlineListToMarkua opts) inlines
+    return $ nowrap contents <> blankline
+
+blockToMarkua _ (CodeBlock attr txt) = do
+    let attributes = attrToMarkua attr
+    let contents = literal txt
+    let shell = literal "```"
+    return $ blankline <> attributes <> cr <> shell <> cr <> contents <> cr 
+                       <> shell <> blankline
+
+-- no attr like format are printed
+blockToMarkua _ (RawBlock _ txt) = do
+    report $ NoTranslation "raw content is writen out in original form and" 
+    return $ cr <> literal txt <> cr 
+
+blockToMarkua opts (BlockQuote blks) = do
+    contents <- blockListToMarkua opts blks
+    let quoteHead = "> "
+    return $ prefixed quoteHead (nestle contents) <> blankline
+
+blockToMarkua opts ((OrderedList (start,style,delim) items)) = do
+    let listAttributes = (start,style,delim)
+    let markers  = orderedListMarkers listAttributes
+    contents <- zipWithM (orderedListItemToMarkua opts) markers items
+    return $ cr <> mconcat contents <> blankline
+
+blockToMarkua opts (BulletList items) = do
+    contents <- mapM (bulletListItemToMarkua opts) items
+    return $ cr <> mconcat contents <> blankline
+    
+blockToMarkua opts (DefinitionList items) = do
+    contents <- mapM (definitionListItemToMarkua opts) items
+    return $ cr <> mconcat contents <> blankline
+
+-- every header gets automatically an attribute list
+blockToMarkua opts (Header level attr inlines) = do
+    contents <- inlineListToMarkua opts inlines
+    let attributes = attrToMarkua attr
+    let eqs = literal $ T.replicate level "#"
+    return $ blankline <> attributes <> cr 
+            <> eqs <> space <> contents <> blankline
+
+blockToMarkua _ HorizontalRule = return $ blankline <> "* * *" <> blankline
+
+-- div is also used to compute the bibliography
+-- if --citeproc and --bibliography is set (div with refs id is generated),
+-- then a bibliography header level 1 with a attr list above is inserted
+blockToMarkua opts (Div attr blks) = do
+    contents <- blockListToMarkua opts blks
+    let attrId = getAttrId attr
+    let attrIdPrefix = T.take 4 attrId
+    let citeId = if attrIdPrefix == "ref-"
+                 then T.drop 4 attrId
+                 else T.empty
+    let headRef = literal (if not $ T.null citeId
+                           then T.cons '[' (T.append citeId "] ")
+                           else T.empty)
+    -- attr only used for bibliography
+    let attributes = attrToMarkua (citeId,[],[])
+    let biblioHeader = literal (if attrId == "refs"
+                                then "{id: bibliography}\n#Bibliography"
```

Comment:
There should probably be a space after `#`, matching other headers. Could we use `blockToMarkua opts (Header ("bibliography",[],[]) "Bibliography"` to generate that part?